src/components/forms/GSIntegerField.jsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import TextField from "@material-ui/core/TextField";
+import GSFormField from "./GSFormField";
+import theme from "../../styles/mui-theme";
+
+export default class GSIntegerField extends GSFormField {
```

Comment:
we can discuss this more later, but I'd like to start moving towards building functional components over class components, especially with modern react, we'll have access to more features (you don't need to make changes for this, just something we can talk about after this upgrade)